Zilon.Core/Zilon.Tournament.ApiGate/Controllers/BotController.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Net.Http.Headers;
+using Zilon.Tournament.ApiGate.BotManagement;
+
+namespace Zilon.Tournament.ApiGate.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BotController : ControllerBase
+    {
+        private static readonly FormOptions DefaultFormOptions = new FormOptions();
+
+        [HttpPost("upload")]
+        public async Task<IActionResult> UploadBotAsync()
+        {
+            var appPath = Environment.GetEnvironmentVariable("APP_PATH");
+            var botRootCatalog = Path.Combine(appPath, "bots");
+
+            if (!MultipartRequestHelper.IsMultipartContentType(Request.ContentType))
+            {
+                return BadRequest($"Expected a multipart request, but got {Request.ContentType}");
+            }
+
+            var boundary = MultipartRequestHelper.GetBoundary(
+                MediaTypeHeaderValue.Parse(Request.ContentType),
+                DefaultFormOptions.MultipartBoundaryLengthLimit);
+            var reader = new MultipartReader(boundary, HttpContext.Request.Body);
+
+            var section = await reader.ReadNextSectionAsync();
+            while (section != null)
+            {
+                var hasContentDispositionHeader = ContentDispositionHeaderValue.TryParse(
+                    section.ContentDisposition,
+                    out var contentDisposition);
+
+                var fileName = contentDisposition.FileName.ToString();
+                if (!IsValidFileName(fileName))
+                {
+                    throw new ArgumentException($"Имя имя '{fileName}' не является корректным именем файла.");
+                }
+
+                if (hasContentDispositionHeader)
+                {
+                    if (MultipartRequestHelper.HasFileContentDisposition(contentDisposition))
+                    {
+
+                        using (var copyStream = new MemoryStream())
+                        {
+                            await section.Body.CopyToAsync(copyStream);
+
+                            copyStream.Seek(0, SeekOrigin.Begin);
+                            var botFullNameAndPath = Path.Combine(botRootCatalog, fileName);
+                            using (var targetStream = System.IO.File.Create(botRootCatalog))
+                            {
+                                await copyStream.CopyToAsync(targetStream);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5310892)